packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -654,62 +678,40 @@ export function onForkTransitive<T extends ISubscribable<{ fork: (t: T) => void
 }
 
 class RevertibleRevision implements Revertible {
-	public readonly origin: Revertible["origin"];
-
 	private referenceCount = 1;
 
 	public constructor(
-		public readonly kind: RevertibleKind,
-		public readonly revision: RevisionTag,
-		private readonly onRevert: (revertible: RevertibleRevision) => void,
-		private readonly onDispose: (revertible: RevertibleRevision) => void,
-	) {
-		this.kind = kind;
-		this.revision = revision;
-		// This is currently always the case, but we may want to support reverting remote ops
-		this.origin = { isLocal: true };
-	}
+		private readonly onRevert: () => void,
+		private readonly onDispose: () => void,
+	) {}
 
 	public get status(): RevertibleStatus {
 		return this.referenceCount === 0 ? RevertibleStatus.Disposed : RevertibleStatus.Valid;
 	}
 
-	public revert(): RevertibleResult {
-		if (this.status === RevertibleStatus.Valid) {
-			this.onRevert(this);
-			// If reverting leads to a schema change then the revertible will be disposed as part of the revert.
-			this.dispose(false);
-			return RevertibleResult.Success;
-		}
-		return RevertibleResult.Failure;
+	public acquire(): RevertibleRevision {
+		assert(this.status === RevertibleStatus.Valid, "cannot acquire a disposed revertible");
+		this.referenceCount += 1;
```

Comment:
we start the count at 1, so is this off by one now? in the previous behavior, we disposed at 1 not 0 (we need tests for this is actually a bug)